design/export-functionality.md

```diff
@@ -0,0 +1,66 @@
+# Export
+
+## Requirements
+The functionality currently provided by the lakeFS Export utility but without a dependency on Apache Spark.
+This includes:
+1. Export all data from a given lakeFS reference (commit or branch) to a designated object store location.
+2. Export only the diff between a specific branch and a commit reference. 
+3. Success/Failure indications on the exported objects. 
+
+
+## Non-Requirements
+1. Support distribution across machines.
+
+
+## Possible Solutions
+
+### Spark client stand-alone
+
+Create a standalone Docker container that runs the spark client. 
+
+Pros:
+- Utilizes the existing spark client, and gives the same functionality. 
+ 
+Cons:
+- Not a built-in command in lakeFS, therefore require some adjustments from the users.
+  - This solution probably requires the greatest effort from the users.
+
+
+### Using the java client
+Duplicate the spark client functionality and create an Exporter class that doesn't use spark. This class can be used with the java client. 
```

Comment:
Do you mean to add an export capability to the generated lakeFS Java client? that is, only to the Java client?

Reply:
Yes